packages/tooltip/src/useTooltip.ts

```diff
@@ -5,39 +5,61 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { useState, useEffect, useRef } from 'react';
+import { useState, useEffect, useRef, ButtonHTMLAttributes } from 'react';
 import { composeEventHandlers, generateId, KEY_CODES } from '@zendeskgarden/container-utilities';
 
-export function useTooltip({ delayMilliseconds = 500, id, isVisible } = {}) {
+export interface IUseTooltipProps {
+  delayMilliseconds?: number;
+  id?: string;
+  isVisible?: boolean;
+}
+
+export interface IUseTooltipReturnValue {
+  isVisible?: boolean;
+  getTooltipProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getTriggerProps: <T>(options?: T) => T & React.ButtonHTMLAttributes<any>;
```

Comment:
We don't use `ButtonHTMLAttributes` for `getTriggerProps` in the other hooks. Is there a reason we are using this instead of `HTMLProps`?

Reply:
> We don't use ButtonHTMLAttributes for getTriggerProps in the other hooks.

It looks like we [use it](https://github.com/zendeskgarden/react-containers/blob/master/packages/modal/src/useModal.ts#L25) for the `useModal` hook.

> Is there a reason we are using this instead of HTMLProps?

`ButtonHTMLAttributes` are used here instead of `HTMLProps` because a button element is used as the trigger element in our Storybook example:

```tsx
<button {...getTriggerProps({ ref: triggerRef })}>Trigger</button>
```

I don't think consumers of `getTriggerProps` **have** to use a `<button>` element. They could use a `<div>`.

Maybe it is better if the type annotation for `getTriggerProps` returns `T & React.HTMLProps<any>`, and then the consumer can cast the prop getter return value where it is used if they use a `<button>`. I think that would work too: 

Type for prop getter:
```ts
getTriggerProps: <T>(options?: T) => T & React.HTMLProps<any>;
```

Then in Storybook example:

```tsx
<button {...getTriggerProps({ ref: triggerRef }) as any}>Trigger</button>
```

Reply:
My vote is for not using `ButtonHTMLAttributes` since it may not always be applied to a `<button>` element. But I'll leave it up to you to decide if the change is worth it.

Reply:
I agree, I will get ahead and make that change so that `HTMLProps` is used rather than `ButtonHTMLAttributes`. Thank you for bringing this up!